src/Enum.php

```diff
@@ -363,31 +364,35 @@ private static function detectConstants($class)
                 $constants = $scopeConstants + $constants;
             } while (($reflection = $reflection->getParentClass()) && $reflection->name !== __CLASS__);
 
-            // Detect ambiguous values and report names
-            $ambiguous = [];
-            foreach ($constants as $value) {
-                $names = \array_keys($constants, $value, true);
-                if (\count($names) > 1) {
-                    $ambiguous[\var_export($value, true)] = $names;
-                }
-            }
-            if (!empty($ambiguous)) {
-                throw new LogicException(
-                    'All possible values needs to be unique. The following are ambiguous: '
-                    . \implode(', ', \array_map(function ($names) use ($constants) {
-                        return \implode('/', $names) . '=' . \var_export($constants[$names[0]], true);
-                    }, $ambiguous))
-                );
-            }
-
             self::$constants[$class] = $constants;
             self::$names[$class] = \array_keys($constants);
+
+            assert('self::assertNonAmbiguous($class)');
         }
 
         return self::$constants[$class];
     }
 
     /**
+     * Assert that the given enumeration class doesn't define ambiguous enumerator values
+     * @param string $class
+     * @return bool
+     */
+    private static function assertNonAmbiguous($class)
+    {
+        $constants = self::$constants[$class];
+        $ambiguous = [];
+        foreach ($constants as $value) {
+            $names = \array_keys($constants, $value, true);
+            if (\count($names) > 1) {
+                $ambiguous[\var_export($value, true)] = $names;
+            }
+        }
+
+        return empty($ambiguous);
```

Comment:
Why not throw an exception here like before? I am kind of against using assertions.

Reply:
I also see not why this should be so much faster.